content/architecture/storage-engine/commit-graph.md

```diff
@@ -0,0 +1,105 @@
+
+## The Init Commit
+
+To create a commit graph you must "initialize" one. Initialization can be done with the `init` command via the command line. This creates an "init commit". In Dolt, `create database` also creates an init commit if you are in the SQL context. 
+
+The `init` command creates a commit with metadata taken from the environment and empty contents.
+
+![init commit](../../.gitbook/assets/commit-graph-init.png)
+
+The init commit is made on the default branch, usually named `main`. A branch is a pointer to a commit. The tip of a branch has a special name or reference called `HEAD`.
+
+![init commit on main](../../.gitbook/assets/commit-graph-main-branch.png)
+
+## `WORKING` and `STAGED`
+
+In Git and Dolt, at the `HEAD` of a branch there are two additional special references, called `STAGED` ad `WORKING`. These references point to active changes you are making to the `HEAD` of the branch. If there are no changes, the contents of `HEAD`, `STAGED`, and `WORKING` are the same. 
+
+![HEAD, STAGED, and WORKING](../../.gitbook/assets/commit-graph-staged-working.png)
+
+When you make changes to the content of a branch, changes are made in `WORKING`. Changes to the content of `WORKING` change its content address.
+
+![WORKING changes](../../.gitbook/assets/commit-graph-working-changes.png)
+
+When you are ready to make a commit, you stage the changes using the `add` command. If you stage all your changes, `STAGED` and `WORKING` point to the same content and thus share the same content address.
+
+![STAGED changes](../../.gitbook/assets/commit-graph-staged-changes.png)
+
+`STAGED` and `WORKING` allow for changes to content to be tested and verified before being stored permanently in the commit graph.
+
+`WORKING` is often called the working set. An interesting way to think about the working set is traditional file systems that don't use Git only have a working set. Traditional databases like MySQL or Postgres only have a working set. If you create a database in Dolt and only run traditional SQL, your working set will look and act exactly like a MySQL database.
+
+## History
+
+Commits are created using the aptly named `commit` command. When you commit your `STAGED` changes, the content that is staged is moved to the tip of the branch and you have an opportunity to add metadata like message and author. The `HEAD` of the branch `main` becomes this newly created commit. 
+
+![A Second Commit](../../.gitbook/assets/commit-graph-second-commit.png)
+
+Commits have zero to many parents. The init commit has zero parents. A normal commit has one parent, representing the previous commit metadata and content. A merge commit, which we'll discuss later, has many parents. 
+
+Parents allow for the history of branches to be computed by walking the branch from its `HEAD`. This is commonly called the commit log and generated using the `log` command.
+
```

Comment:
Could be worth giving the example output of running `log` on `main` at this point. Like "Using the log command on the 'main' branch here would list commits `h512kl` and `t1ms3n`."